kafka-producer-java/src/main/java/Main.java

```diff
@@ -9,14 +9,24 @@ public static void main(String[] args) throws Exception {
         System.out.println("config init");
         Monitor.init();
         System.out.println("monitor init");
-        producer = new Producer(config, monitor).start();
+        producer = createProducer(config);
         System.out.println("producer started");
         server = new Server(config, monitor, producer).start();
         System.out.println("server started");
         Runtime.getRuntime().addShutdownHook(new Thread(() -> close()));
         Monitor.started();
     }
 
+    private static Producer createProducer(Config config) {
```

Comment:
you're hiding that you're starting the producer, not just creating it.
this method should only map the config to the producer type. the start should be outside of the method scope.